salt/scripts.py

```diff
@@ -94,7 +94,10 @@ def suicide_when_without_parent(parent_pid):
         random_delay = randint(1, delay)
         log.info('Sleeping random_reauth_delay of {0} seconds'.format(random_delay))
         # preform delay after minion resources have been cleaned
-        queue.put(random_delay)
+        if minion.options.daemon:
+            salt_minion()
```

Comment:
Sleeping here the random delay and then call start_minion to get the same behaviour?